### ISSUE TITLE
Dropdown menus rendering "under" the editor region in latest Safari on macOS

### ISSUE THREAD

**Issue by gtritchie** [rstudio/rstudio#10821]
### System details

    RStudio Edition : Server
    RStudio Version : Probably all, I only tested `main`
    OS Version      : macOS 12.3 with Safari 15.4
    R Version       : Any

### Steps to reproduce the problem

- update to macOS 12.3 with Safari 15.4
- load RStudio Server in Safari
- open a document (or create a new one, etc.)
- try to use the main menu, e.g. click Help, or one of the dropdowns in the editor toolbar

### Describe the problem in detail

- the dropdown displays under the editor pane

![Screen Shot 2022-03-22 at 09 36 57](https://user-images.githubusercontent.com/10569626/159520787-e4d46ff8-2fa3-4764-9cf8-5678ebda02c0.png)

### Describe the behavior you expected

- dropdown menus display as they always have
- note they work fine on other browsers such as Chrome on the same macOS

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by heath-r** [comment]
Here are two tickets that have reported the issue at this point:
https://rstudioide.zendesk.com/agent/tickets/72850
https://rstudioide.zendesk.com/agent/#/tickets/73054

**Comment by kevinushey** [comment]
This issue appears to be more than just a simple z-index issue; the menu appears to come into view if you try to mouse over the menu after it is displayed. For example:

https://user-images.githubusercontent.com/1976582/159782418-45a462bd-1196-4f9d-9b6b-32e5ff27b387.mov

We'll probably need some kind of gnarly CSS hack to tweak the way Safari tries to render these things, if we need a workaround.

**Comment by shrektan** [comment]
Hi, 

I noticed two issues on the latest macOS, one is reported in this issue, the other is that the left part of the pop-up help panel can't be displayed properly on Chrome and Edge (see the screenshot below). The only browser that RSW can work is Firefox, at the present. 

I'm wondering if PR #10832 will resolve the pop-up help panel issue for Chrome and Edge on macOS as well?

<img width="614" alt="image" src="https://user-images.githubusercontent.com/8368933/160040337-bd5ace92-7ff0-4267-8036-e8693b2cc112.png">

**Comment by kevinushey** [comment]
Thanks @shrektan -- I think this is a different issue, so I've opened that separately. https://github.com/rstudio/rstudio/issues/10841

**Comment by ashwinikalantri** [comment]
Hello @kevinushey, can you update when this fix will be available? The UI glitches are making using RStudio maddening.

**Comment by ashwinikalantri** [comment]
As I have already submitted in my [ticket](https://rstudioide.zendesk.com/agent/tickets/72850), even the prompts for objects don't get rendered. They are there and selectable, but not visible.

https://user-images.githubusercontent.com/4697107/160247648-dd5cb37d-8aa2-48d8-9b22-ac51210506b2.mov

**Comment by kevinushey** [comment]
> can you update when this fix will be available?

We're working on it, but to be clear, this is a bug in Safari, so if you have some ire to direct I'd recommend directing it towards Apple (via a bug report or similar).

**Comment by shrektan** [comment]
Yes, I doubt it's a bug of Safari, as RStudio Workbench is not the only app that's not working after upgrading to the latest Safari. But I don't know how to report it to Apple and how to make them recognize this is a bug of their own.

**Comment by ronblum** [comment]
@mikebessuille & @kevinushey Do we want to backport this to Ghost Orchid, as well?

Problem reproduced on Ubuntu 20.04 in RStudio Server 
- Ghost Orchid 2021.09.2-382
- Prairie Trillium 2022.02.0-443
- Spotted Wakerobin 2022.06.0-daily+225

Verified fixed on Ubuntu 20.04 in RStudio Server
- Prairie Trillium 2022.02.2+481
- Spotted Wakerobin 2022.06.0-daily+284

File type matters. This isn't an issue with R scripts, but it is with R Markdown, C++, Shiny, and others. Tested by creating the following file types:
- R script (baseline—never a problem)
- R Markdown
- Shiny
- C++

Confirmed that this is fixed in Safari 15.4 on MacOS 12.3. Also confirmed that this didn't affect Chrome and Safari.

**Comment by mikebessuille** [comment]
I'd be okay with patching this to Ghost Orchid as well.

**Comment by kevinushey** [comment]
Fix has now been backported to Ghost Orchid as well.

**Comment by ronblum** [comment]
Verified in RStudio Server 2021.09.3-392 (Ghost Orchid) on Ubuntu 20.04
- Fixed for Safari on MacOS 12.3
- Continues to work in Chrome and Firefox on MacOS 12.3

**Comment by ashwinikalantri** [comment]
<img width="651" alt="Studio" src="https://user-images.githubusercontent.com/4697107/174491665-6433adb1-b6ee-4653-b268-5e903b62c7d4.png">

Most of the issue was resolved, the drop down menus, the function autocomplete are rendered correctly. But the tool tip that shows the function and attributes needed (see image above) is incorrectly rendered.

I am running RStudio Workbench 2022.02.3+492.pro3 on Safari 15.5 (Mac OS 12.4)

**Comment by kevinushey** [comment]
We probably need to include `gwt-PopupPanel` here as well:

https://github.com/rstudio/rstudio/pull/10832/files?w=1#diff-f2f2622845712c10ea99db19d8ca19c14a77872bfa88fc5f9f96fe56c0e7b9adR29

**Comment by timtmok** [comment]
Kevin's change was merged after this issue was created and does fix the problem. I'm unable to reproduce on Safari 15.5. I can see that the body element has the `safari` class which causes the menu element to inherit the transform property that force the element to the top of the Z layer.

That change was backported to `prairie-trillium` and exists in `main` now.

**Comment by timtmok** [comment]
Actually, I'll reopen for now because I don't think this was backported to Ghost Orchid.

**Comment by timtmok** [comment]
I've reproduced the issue and merged the change to PT and GO branches.

**Comment by ronblum** [comment]
Is this for the drop down menus, or one of the tooltip problems?

**Comment by timtmok** [comment]
> Is this for the drop down menus, or one of the tooltip problems?

This change fixes the tooltip. The drop down menu fix was merged back in March so that is available all the way back to Ghost Orchid.

**Comment by timtmok** [comment]
> The search box in the help window also renders the auto-complete behind other elements.
I've split this off as its own issue. Any other issues with Safari compatibility can be a new issue so it can be tracked independently.

**Comment by ronblum** [comment]
@timtmok Thanks! I've follow up on the tooltip issue.

**Comment by ronblum** [comment]
Tool tips issues verified in RStudio Server 2022.07.0-preview+543 Spotted Wakerobin on MacOS 12.5. Verified using
* Safari 15.5

Plus checked other browser to make sure that there are no regressions:
* Chrome 103.0.5060.53
* Safari 101.0.1

I'm leaving this open for testing the backports to Ghost Orchid and Prairie Trillium.
